Tests/SwiftParserTest/translated/SubscriptingTests.swift

```diff
@@ -447,26 +461,18 @@ final class SubscriptingTests: XCTestCase {
       }
       """,
       diagnostics: [
-        // TODO: Old parser expected error on line 2: expected '(' for subscript parameters
         DiagnosticSpec(locationMarker: "1️⃣", message: "expected '(' to start parameter clause"),
         DiagnosticSpec(locationMarker: "2️⃣", message: "expected '(' to start function type"),
         DiagnosticSpec(locationMarker: "3️⃣", message: "expected ')' in function type"),
         DiagnosticSpec(locationMarker: "4️⃣", message: "expected ')' to end parameter clause"),
         DiagnosticSpec(locationMarker: "4️⃣", message: "expected '->' and return type in subscript"),
-      ]
-    )
-  }
-
-  func testSubscripting25() {
-    AssertParse(
+      ], fixedSource: """
+      struct A11 {
+        subscript( x y : (Int) -> Int)  -> <#type#>{
```

Comment:
Woaw. That's some pretty crazy (and arguably incorrect) recovery. I'm not sure how we could end up with the "ideal" recovery of `subscript(x y: Int) -> Int` though, we're always going to eat that `->` as part of the type. Given how unlikely this case is to come up, it's probably fine.

Reply:
Yeah, I think getting the correct recovery would require a lot more context. Basically, we would need to tell type parsing that it shouldn’t parse the last `->`, but that would require a lot of lookahead. Also, I’m not sure if I care too much. This code is pretty broken.

Reply:
Yeah I'm definitely of the same opinion 👍. I can't see someone actually writing this. As long as we produce *something* decl related here I'd say it's fine.

Reply:
What's much more likely is probably `subscript(x y: Int -> Int {`, ie. they just forgot the final parenthesis. But yeah, I agree it would require too much lookahead.